l10n_it_reverse_charge/models/account_move.py

```diff
@@ -112,7 +114,7 @@ def rc_inv_vals(self, partner, rc_type, lines, currency):
             "Internal reference: %(internal_reference)s",
             supplier=supplier.display_name,
             reference=self.invoice_origin or self.ref or "",
-            date=self.date,
+            date=reference_date,
             internal_reference=self.name,
```

Comment:
Se il riferimento interno è di `self`, mi aspetto che anche la data venga presa da `self`: quindi secondo me la scelta sta tra `self.date` e `self.invoice_date` (da decidere in https://github.com/OCA/l10n-italy/issues/4141#issuecomment-2138848750).
Una volta scelto si potrebbe anche chiarire nel testo della descrizione quale delle due date è riportata.

Reply:
Pare che sia stato risolto dal test funzionale. Personalmente dopo mesi di utilizzo (in produzione) non ho ancora riscontrato problemi.